common/src/java/com/zimbra/common/io/TcpServerInputStream.java

```diff
@@ -50,18 +53,24 @@ public TcpServerInputStream(InputStream in, int size) {
      *         the end of the stream has been reached
      * @throws IOException
      */
-    public String readLine() throws IOException {
-        buffer.delete(0, buffer.length());
-        while (true) {
-            int ch = read();
-            if (ch == -1) {
-                return null;
-            } else if (ch == CR) {
-                continue;
-            } else if (ch == LF) {
-                return buffer.toString();
-            }
-            buffer.append((char)ch);
-        }
-    }
+	public String readLine() throws IOException {
+		buffer.delete(0, buffer.length());
+		br = new BufferedReader(new InputStreamReader(this, StandardCharsets.UTF_8));
```

Comment:
I think there is no need to create BufferReader object on each readLine. 
Also we are closing just once.

Move BufferReader to both constructors.

Thanks